### ISSUE TITLE
[DOC]: Add some documentation about which models are implemented within PyCaret, and where to go to read more information about the underlying libraries

### ISSUE THREAD

**Issue by chrimaho** [pycaret/pycaret#3167]
### pycaret version checks

- [X] I have checked that the issue still exists on the latest versions of the docs [here](https://pycaret.gitbook.io/docs/)

### Location of the documentation

https://pycaret.gitbook.io/docs/

### Documentation problem

There's no documentation about the specific models which are implemented within PyCaret.

I know that I can see the list of the names of the models implemented when I run the `compare_models()` function (for example, see [here](https://pycaret.gitbook.io/docs/get-started/quickstart#compare-models-2)). However, this does not actually tell me the details and the sources behind the models. And the doc website currently does not contain this information either.

### Suggested fix for documentation

Please add some documentation about the models which are implemented.

#### Option 1: Add new page

- On the [main page](https://pycaret.gitbook.io/docs/), please add one new option on the left-hand side, under the [📶 Modules](https://pycaret.gitbook.io/docs/get-started/modules) button, please add a new button called [🎁 Models](#). This new button should point to the page at URL https://pycaret.gitbook.io/docs/get-started/models
- This new Models page (https://pycaret.gitbook.io/docs/get-started/modules) should have a table with one row per model. One suggestion for how the table could look is given in the below table.
- The headers for this table should allow for filtering, to make it easy for the user to focus only on the models they require.

| Category | Classification | Model Name | Library Utilised | Underlying Class | More Info |
|---|---|---|---|---|---|
| Supervised | Regression | Linear Regression | SkLearn | `LinearRegressor` | [source](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html) |
| Supervised | Regression | Light Gradient Boosting Machine | LightGBM | `LGBMRegressor` | [source](https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.LGBMRegressor.html) |
| Supervised | Classification | Logistic Regression | SkLearn | `LogisticRegression` | [source](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html) |
| Supervised | Classification | Extreme Gradient Boosting | XGBoost | `XGBClassifier` | [source](https://xgboost.readthedocs.io/en/stable/python/python_api.html#xgboost.XGBClassifier) |
| Unsupervised | Clustering | Spectral Clustering | SkLearn | `SpectralClustering` | [source](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.SpectralClustering.html) |
| Unsupervised | Clustering | K-Means Clustering | cuML | `KMeans` | [source](https://docs.rapids.ai/api/cuml/stable/api.html#cuml.KMeans) |
| Time Series | | Auto ARIMA | SkTime | `AutoARIMA` | [source](https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.arima.AutoARIMA.html#autoarima) |
| Time Series | | CatBoost Regressor | CatBoost | `CatBoostRegressor` | [source](https://catboost.ai/en/docs/concepts/python-reference_catboostregressor) |

#### Option 2: Extend the existing [📶 Modules](https://pycaret.gitbook.io/docs/get-started/modules) page

- When navigating to the [📶 Modules](https://pycaret.gitbook.io/docs/get-started/modules) page, this is already separated by Supervised/Unsupervised/Time Series/etc; and under each of these are further segregations by Regression/Classification/Clustering/etc.
- Each of these groups currently contains the three selections for:
    - Quickstart
    - API Docs
    - Tutorial
- I propose to add one more option to this list for 'Models'. Therefore, under each group, the list would look like this:
    - Quickstart
    - API Docs
    - Models
    - Tutorial
- This 'Models' button would pull up a page which has a list of each of the Models which are implemented, including the source package, underlying Class name, and also where to go to read more information about the model from the source library.

**Comment by chrimaho** [comment]
This is an important topic. Can you please keep it open?

**Comment by ngupta23** [comment]
Removed the no-activity tag.

**Comment by chrimaho** [comment]
Please re-open this issue @ngupta23

**Comment by PabloJMoreno** [comment]
I can help here

**Comment by chrimaho** [comment]
> Stale issue message: This issue will be automatically closed by GitHub Actions in 1 week if there is no further activity.

Again??

@ngupta23 or @PabloJMoreno, are you guys looking in to this one?

**Comment by ngupta23** [comment]
I am not looking at this. Any help on documenting this would be appreciated.

**Comment by chrimaho** [comment]
I'm working on it.
